src/box/lua/schema.lua

```diff
@@ -1619,6 +1620,52 @@ local function func_id_by_name(func_name, level)
 end
 box.internal.func_id_by_name = func_id_by_name -- for space.upgrade
 
+local function update_index_opts(format, opts, level)
+    if opts.func ~= nil and type(opts.func) == 'string' then
+        opts.func = func_id_by_name(opts.func, 2)
+    end
+    -- Do not much care if opts.covers is something strange like
+    -- sparse array or map with keys.
+    local covers = {}
+    local field_presence = {}
+    for i, field in ipairs(opts.covers or {}) do
+        local idx
+        local field_ref = "options.covers[" .. i .. "]: "
+        if type(field) == 'string' then
+            idx = format_field_index_by_name(format, field)
+            if idx == nil then
+                box.error(box.error.ILLEGAL_PARAMS, field_ref ..
+                          "field was not found by name '" .. field .. "'",
+                          level + 1)
+            end
+        elseif type(field) == 'number' then
+            if field <= 0 then
+                box.error(box.error.ILLEGAL_PARAMS, field_ref ..
+                          "field (number) must be one-based", level + 1)
```

Comment:
This check should be done somewhere in `alter.cc`. Same about duplicate fields.

Reply:
Please, check the previous conversation.